test/sql/function/string/regex_extract_all.test

```diff
@@ -0,0 +1,670 @@
+  VALUES ('XY'),('X1Y'),('X123Y'),('X1Y23')
+) AS t(str)
+----
+XY	[XY]
+X1Y	[X1Y]
+X123Y	[]
+X1Y23	[X1Y2]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'ab*cd') AS matched
+FROM (
+  VALUES ('acd'), ('abcd'), ('abcdacd'), ('abbcd'), ('abbbcd'), ('ab1cd')
+) AS t(str)
+----
+acd	[acd]
+abcd	[abcd]
+abcdacd	[abcd, acd]
+abbcd	[abbcd]
+abbbcd	[abbbcd]
+ab1cd	[]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'a(bc)*d') AS matched
+FROM (
+  VALUES ('ad'), ('abd'), ('acd'), ('abcd'), ('abcxd'), ('abcbcd'), ('abcbce')
+) AS t(str)
+----
+ad	[ad]
+abd	[]
+acd	[]
+abcd	[abcd]
+abcxd	[]
+abcbcd	[abcbcd]
+abcbce	[]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'a[bc]*d') AS matched
+FROM (
+  VALUES ('ad'), ('abd'), ('acd'), ('abcd'), ('abcbbcd'), ('abce')
+) AS t(str)
+----
+ad	[ad]
+abd	[abd]
+acd	[acd]
+abcd	[abcd]
+abcbbcd	[abcbbcd]
+abce	[]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'X[\d]*Y[\d]*') AS matched
+FROM (
+  VALUES ('XY'),('X1Y'),('X123Y'),('X1Y23'),('X12Z34Y')
+) AS t(str)
+----
+XY	[XY]
+X1Y	[X1Y]
+X123Y	[X123Y]
+X1Y23	[X1Y23]
+X12Z34Y	[]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'ab+cd') AS matched
+FROM (
+  VALUES ('acd'), ('abcd'), ('abcdacd'), ('abbcd'), ('abbbcd'), ('ab1cd')
+) AS t(str)
+----
+acd	[]
+abcd	[abcd]
+abcdacd	[abcd]
+abbcd	[abbcd]
+abbbcd	[abbbcd]
+ab1cd	[]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'(ab)+cd') AS matched
+FROM (
+  VALUES ('bcd'), ('abcd'), ('abbcd'), ('ababcd'), ('ababxcd')
+) AS t(str)
+----
+bcd	[]
+abcd	[abcd]
+abbcd	[]
+ababcd	[ababcd]
+ababxcd	[]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'a[bc]+d') AS matched
+FROM (
+  VALUES ('ad'), ('abd'), ('acd'), ('abcd'), ('abce'), ('abcbd')
+) AS t(str)
+----
+ad	[]
+abd	[abd]
+acd	[acd]
+abcd	[abcd]
+abce	[]
+abcbd	[abcbd]
+
+query II
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'X[\d]+Y[\d]*') AS matched
+FROM (
+  VALUES ('XY'),('X1Y'),('X123Y'),('X1Y23'),('X12Z34Y')
+) AS t(str)
+----
+XY	[]
+X1Y	[X1Y]
+X123Y	[X123Y]
+X1Y23	[X1Y23]
+X12Z34Y	[]
+
+query IIIII
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'ab{3}cd')   AS m1,
+  REGEXP_EXTRACT_ALL(str,'ab{1,}cd')  AS m2,
+  REGEXP_EXTRACT_ALL(str,'ab{1,2}cd') AS m3,
+  REGEXP_EXTRACT_ALL(str,'ab{0,2}cd') AS m4
+FROM (
+  VALUES ('acd'), ('abcd'), ('abcdacd'), ('abbcd'), ('abbbcd'), ('ab1cd')
+) AS t(str)
+----
+acd	[]	[]	[]	[acd]
+abcd	[]	[abcd]	[abcd]	[abcd]
+abcdacd	[]	[abcd]	[abcd]	[abcd, acd]
+abbcd	[]	[abbcd]	[abbcd]	[abbcd]
+abbbcd	[abbbcd]	[abbbcd]	[]	[]
+ab1cd	[]	[]	[]	[]
+
+query IIIII
+SELECT str, 
+  REGEXP_EXTRACT_ALL(str,'(ab){2}cd')   AS m1,
+  REGEXP_EXTRACT_ALL(str,'(ab){1,}cd')  AS m2,
+  REGEXP_EXTRACT_ALL(str,'(ab){1,2}cd') AS m3,
+  REGEXP_EXTRACT_ALL(str,'(ab){0,2}cd') AS m4
+FROM (
+  VALUES ('acd'), ('bcd'),('abcd'), ('abbcd'), ('ababcd'), ('ab1cd')
+) AS t(str)
+----
+acd	[]	[]	[]	[cd]
+bcd	[]	[]	[]	[cd]
+abcd	[]	[abcd]	[abcd]	[abcd]
+abbcd	[]	[]	[]	[cd]
+ababcd	[ababcd]	[ababcd]	[ababcd]	[ababcd]
+ab1cd	[]	[]	[]	[cd]
+
+query IIIIII
```

Comment:
Can `regexp_extract_all` also support the re2 modifiers (e.g. `'i'` for case-insensitivity, etc)